### PR TITLE
Register all known properties in format 3

### PR DESCRIPTION
**Pull request by IvanUkhov**
This is to incorporate Georg's property list in #881.

### CODE REVIEW

File: Lib/glyphsLib/builder/custom_params.py [hunk 1]
```diff
@@ -367,57 +368,67 @@ def register(handler):
...
+    ("postscriptDefaultWidthX", None, None),
+    ("postscriptNominalWidthX", None, None),
+    ("postscriptWeightName", None, None),
+    ("postscriptDefaultCharacter", None, None),
+    ("postscriptWindowsCharacterSet", None, None),
+    ("macintoshFONDFamilyID", None, None),
+    ("macintoshFONDName", None, None),
+    ("styleMapFamilyName", None, None),
+    ("styleMapStyleName", None, None),
+    # TODO: Map these properties to custom parameters.
```
Comment (IvanUkhov) [inline Lib/glyphsLib/builder/custom_params.py:407]: I did not fine place for these in what is listed above. Please let me know.

File: Lib/glyphsLib/builder/custom_params.py [hunk 2]
```diff
@@ -367,57 +368,67 @@ def register(handler):
...
+    ("postscriptDefaultWidthX", None, None),
+    ("postscriptNominalWidthX", None, None),
+    ("postscriptWeightName", None, None),
+    ("postscriptDefaultCharacter", None, None),
+    ("postscriptWindowsCharacterSet", None, None),
+    ("macintoshFONDFamilyID", None, None),
+    ("macintoshFONDName", None, None),
+    ("styleMapFamilyName", None, None),
+    ("styleMapStyleName", None, None),
+    # TODO: Map these properties to custom parameters when applicable.
```
Comment (IvanUkhov) [inline Lib/glyphsLib/builder/custom_params.py:407]: I could not find place for these in the above.

File: Lib/glyphsLib/builder/custom_params.py [hunk 3]
```diff
@@ -310,52 +310,53 @@ def register(handler):
...
-    ("hheaLineGap", "openTypeHheaLineGap"),
-    ("compatibleFullName", "openTypeNameCompatibleFullName"),
-    ("WWSSubfamilyName", "openTypeNameWWSSubfamilyName"),
+GLYPHS_UFO_MAPPING = (
+    # (Glyphs custom parameter, UFO custom parameter, Glyphs property)
+    ("hheaAscender", "openTypeHheaAscender", None),
+    ("hheaDescender", "openTypeHheaDescender", None),
+    ("hheaLineGap", "openTypeHheaLineGap", None),
+    ("compatibleFullName", "openTypeNameCompatibleFullName", None),
+    ("WWSSubfamilyName", "openTypeNameWWSSubfamilyName", "WWSSubfamilyName"),
```
Comment (IvanUkhov) [inline Lib/glyphsLib/builder/custom_params.py:319]: This is the only actual change.

**Comment by IvanUkhov** [comment]
@anthrotype, and here are the missing properties.

File: Lib/glyphsLib/builder/custom_params.py [hunk 4]
```diff
@@ -310,52 +310,53 @@ def register(handler):
     KNOWN_PARAM_HANDLERS.append(handler)
 
 
-GLYPHS_UFO_CUSTOM_PARAMS = (
-    ("hheaAscender", "openTypeHheaAscender"),
-    ("hheaDescender", "openTypeHheaDescender"),
-    ("hheaLineGap", "openTypeHheaLineGap"),
-    ("compatibleFullName", "openTypeNameCompatibleFullName"),
-    ("WWSSubfamilyName", "openTypeNameWWSSubfamilyName"),
+GLYPHS_UFO_MAPPING = (
```
Comment (anthrotype) [inline Lib/glyphsLib/builder/custom_params.py:313]: why did you merge those three lists of tuples into one? I don't particularly find a single list (with a bunch of Nones) more readable...

Comment (anthrotype) [reply Lib/glyphsLib/builder/custom_params.py:313]: The problem is not the Bones, the different lists grouped similar items that received similar treatment, which a single list obfuscates a bit. I don't mind the git diff changes.

File: Lib/glyphsLib/builder/custom_params.py [hunk 5]
```diff
@@ -311,11 +311,13 @@ def register(handler):
 
 
 GLYPHS_UFO_CUSTOM_PARAMS = (
+    # These are be stored in the official descriptor attributes.
+    # "familyName",
+    # "fileName",
+    ("compatibleFullName", "openTypeNameCompatibleFullName"),
```
Comment (IvanUkhov) [inline Lib/glyphsLib/builder/custom_params.py:317]: Just moved up to group names.

File: Lib/glyphsLib/builder/custom_params.py [hunk 6]
```diff
@@ -311,11 +311,13 @@ def register(handler):
...
     ("hheaAscender", "openTypeHheaAscender"),
     ("hheaDescender", "openTypeHheaDescender"),
     ("hheaLineGap", "openTypeHheaLineGap"),
-    ("compatibleFullName", "openTypeNameCompatibleFullName"),
-    ("WWSSubfamilyName", "openTypeNameWWSSubfamilyName"),
```
Comment (IvanUkhov) [inline Lib/glyphsLib/builder/custom_params.py:318]: Moved to the list with properties.

File: Lib/glyphsLib/builder/custom_params.py [hunk 7]
```diff
@@ -355,23 +359,50 @@ def register(handler):
 for glyphs_name, ufo_name in GLYPHS_UFO_CUSTOM_PARAMS:
     register(ParamHandler(glyphs_name, ufo_name, glyphs_long_name=ufo_name))
 
+# Reference:
+# https://github.com/googlefonts/glyphsLib/pull/881#issuecomment-1474226616
 GLYPHS_UFO_CUSTOM_PARAMS_GLYPHS3_PROPERTIES = (
+    # This is stored in the official descriptor attributes.
```
Comment (IvanUkhov) [inline Lib/glyphsLib/builder/custom_params.py:365]: Reordered to keep in sync with `constants.PROPERTIES_WHITELIST`.

**Review by anthrotype — APPROVED** [review]

**Pull request merged**